.github/workflows/ci.yml

```diff
@@ -95,18 +95,34 @@ jobs:
       run: cmake --build $GITHUB_WORKSPACE/build --config ${{ matrix.type.name == 'Debug' && 'Debug' || 'Release' }} --target install
 
     - name: Test
-      if: runner.os == 'Windows'
-      shell: bash
-      run: cmake --build $GITHUB_WORKSPACE/build --target runtests --config ${{ matrix.type.name == 'Debug' && 'Debug' || 'Release' }}
-
-    - name: Test
-      if: runner.os != 'Windows'
-      shell: bash
-      run: ${{matrix.platform.prefix}} ctest --test-dir $GITHUB_WORKSPACE/build --output-on-failure --config ${{ matrix.type.name == 'Debug' && 'Debug' || 'Release' }}
-
-    - name: Generate Coverage Report
-      if: matrix.type.name == 'Debug' && runner.os != 'Windows' # Coverage is already generated on Windows when running tests.
-      run: gcovr -r $GITHUB_WORKSPACE -x $GITHUB_WORKSPACE/build/coverage.out -s -f 'src/SFML/.*' -f 'include/SFML/.*' ${{ matrix.platform.gcovr_options }} $GITHUB_WORKSPACE
+      uses: nick-fields/retry@v2
+      with:
+        max_attempts: 10
+        timeout_minutes: 3
+        retry_wait_seconds: 5
+        warning_on_retry: false
+        shell: bash
+        command: |
+          if [ "${{ runner.os }}" == "Windows" ]; then
```

Comment:
Any specific reason to move the OS check down into the executed script instead of keeping it on the workflow level? I don't see too much duplicated settings otherwise.

If you keep it that way, should probably use single quotes (`'…'`), since it's static strings.

Reply:
The OS check is performed only once to decide if the step is run or not. In this case we have a second branch in the script to check if we are macOS or Linux. If we wanted to move it out to workflow level we would have 3 Test steps with identical action parameters (which actually occupy more lines than the command itself).